calico.py

```diff
@@ -441,14 +449,87 @@ def _call_ipam_plugin(self, env):
                                        "msg": e.msg,
                                        "details": e.details})
                 code = e.code
+        elif self.ipam_type == "host-local":
+            # We've been told to use the "host-local" IPAM plugin.
+            # Check if we need to use the Kubernetes podCidr for this node, and
+            # if so replace the subnet field with the correct value.
+            if self.network_config["ipam"].get("subnet") == "usePodCidr":
+                if not self.running_under_k8s:
+                    print_cni_error(ERR_CODE_GENERIC, "Invalid network config",
+                            "Must be running under Kubernetes to use \
+                                    'subnet: usePodCidr'")
+                _log.info("Using Kubernetes pod cidr")
+                pod_cidr = self._get_kubernetes_pod_cidr()
+                self.network_config["ipam"]["subnet"] = str(pod_cidr)
+
+            # Call the IPAM plugin.
+            _log.debug("Calling host-local IPAM plugin")
+            code, response = self._call_binary_ipam_plugin(env)
         else:
+            # Using some other IPAM plugin - call it.
             _log.debug("Using binary plugin")
             code, response = self._call_binary_ipam_plugin(env)
 
         # Return the IPAM return code and output.
         _log.debug("IPAM response (rc=%s): %s", code, response)
         return code, response
 
+    def _get_kubernetes_pod_cidr(self):
+        """
+        Attempt to get the Kubernetes pod CIDR for this node.
+        First check if we've written it to disk.  If so, use that value.  If
+        not, then query the Kubernetes API for it.
+        """
+        local_store_path = os.path.abspath("podCidr.cache")
```

Comment:
Where are we documenting that the our CNI plugin now _sometimes_ needs to be able to write to disk?